### PR TITLE
ISSD-1454 [PRM] Add 'Cancel' option to MDF request listings' kebab menu for CMM and CMD

### PR DESCRIPTION
**Pull request by abfranca**

### CODE REVIEW

**Comment by abfranca** [comment]
ci:test:sf

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx [hunk 1]
```diff
@@ -83,6 +86,47 @@ export default function getMDFListColumns(
...
+							`/o/${LiferayAPIs.OBJECT}/${ResourceName.MDF_REQUEST_DXP}/${row[MDFColumnKey.ID]}?nestedFields=mdfReqToActs%2C%20mdfReqToMDFClms`
+					);
+					previousValue.push({
+						icon: 'block',
+						key: Status.CANCELED.key,
+						label: ' Cancel',
+						onClick: () => {
+							Liferay.Util.openConfirmModal({
+								message:
+									'Are you sure to cancel the MDF request?',
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx:103]: I think this should read as "Are you sure you want to cancel the MDF request?". "Are you sure to cancel" isn't grammatically correct.

Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx:103]: hey Kyle! i've just used the text that is on the ticket. should i ask Aryan about this?

Comment (kylembischof) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx:103]: Yes, please ask about it on the ticket then. Thanks!

Comment (jairmedeiros) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx:103]: Hey @abfranca, you could ask to Caio Monteiro.

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/getDropdownOptions.ts [hunk 2]
```diff
@@ -6,13 +6,23 @@
...
+import { Liferay } from '../../../common/services/liferay';
 import {Status} from '../../../common/utils/constants/status';
 
 export default function getDropdownOptions(
 	actions: PermissionActionType[],
 	mdfRequestStatus: LiferayPicklist,
 	updateRequestStatus: (status: LiferayPicklist) => Promise<void>
 ) {
+	const callConfirmCancelMDFRequestModal = () => Liferay.Util.openConfirmModal({
+		message: 'Are you sure to cancel the MDF request?',
```
[the remark this answers is not in the document]
Comment (jairmedeiros) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/getDropdownOptions.ts:18]: Hey @abfranca, you could ask to Caio Monteiro.

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx (hunk 1 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/getDropdownOptions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/getDropdownOptions.ts]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts [hunk 3]
```diff
@@ -14,7 +15,8 @@ import {Status} from '../../../common/utils/constants/status';
 const patchRequestStatus = async (
 	mdfRequestStatus: LiferayPicklist,
 	mdfRequestId: string,
-	mdfReqToActs?: MDFRequestActivityDTO[]
+	mdfReqToActs?: MDFRequestActivityDTO[],
+	mdfReqToMDFClms?: MDFClaimDTO[]
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts:19]: Maybe you can do the request inside this patchRequestStatus function.

```suggestion
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts [hunk 4]
```diff
@@ -31,18 +33,63 @@ const patchRequestStatus = async (
...
-					activity.id &&
-						activity.activityStatus?.key === Status.SUBMITTED.key &&
+					if (activity.id && (activity.activityStatus?.key === Status.SUBMITTED.key || activity.activityStatus?.key === Status.CANCELED.key)) {
 						(await patchObjectEntry(
 							ResourceName.ACTIVITY_DXP,
 							activity.id,
 							{
 								activityStatus: Status.APPROVED,
 							}
 						));
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts:43]:
```suggestion
						await patchObjectEntry(
							ResourceName.ACTIVITY_DXP,
							activity.id,
							{
								activityStatus: Status.APPROVED,
							}
						);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts [hunk 5]
```diff
@@ -31,18 +33,63 @@ const patchRequestStatus = async (
...
+				if (mdfReqToMDFClms?.length) {
+					for (const claim of mdfReqToMDFClms) {
+						if (claim.id && claim.mdfClaimStatus?.key === Status.CANCELED.key) {
+							(await patchObjectEntry(
+								ResourceName.MDF_CLAIM_DXP,
+								claim.id,
+								{
+									mdfClaimStatus: Status.APPROVED,
+								}
+							));
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts:56]:
```suggestion
							await patchObjectEntry(
								ResourceName.MDF_CLAIM_DXP,
								claim.id,
								{
									mdfClaimStatus: Status.APPROVED,
								}
							);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts [hunk 6]
```diff
@@ -31,18 +33,63 @@ const patchRequestStatus = async (
...
+			) {
+				for (const activity of mdfReqToActs) {
+					if (activity.id && activity.activityStatus?.key === Status.APPROVED.key) {
+						(await patchObjectEntry(
+							ResourceName.ACTIVITY_DXP,
+							activity.id,
+							{
+								activityStatus: Status.CANCELED,
+							}
+						));
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts:75]:
```suggestion
						await patchObjectEntry(
							ResourceName.ACTIVITY_DXP,
							activity.id,
							{
								activityStatus: Status.CANCELED,
							}
						);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts [hunk 7]
```diff
@@ -31,18 +33,63 @@ const patchRequestStatus = async (
...
+				if (mdfReqToMDFClms?.length) {
+					for (const claim of mdfReqToMDFClms) {
+						if (claim.id && (claim.mdfClaimStatus?.key === Status.APPROVED.key || claim.mdfClaimStatus?.key === Status.DRAFT.key)) {
+							(await patchObjectEntry(
+								ResourceName.MDF_CLAIM_DXP,
+								claim.id,
+								{
+									mdfClaimStatus: Status.CANCELED,
+								}
+							));
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts:88]:
```suggestion
							await patchObjectEntry(
								ResourceName.MDF_CLAIM_DXP,
								claim.id,
								{
									mdfClaimStatus: Status.CANCELED,
								}
							);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx [hunk 8]
```diff
@@ -83,6 +86,47 @@ export default function getMDFListColumns(
...
+								message:
+									'Are you sure you want to cancel the MDF request?',
+								onConfirm: async (isConfirmed: boolean) => {
+									if (isConfirmed) {
+										const newRequestStatus = await patchRequestStatus(
+											Status.CANCELED,
+											String(row[MDFColumnKey.ID]),
+											mdfRequest?.mdfReqToActs,
+											mdfRequest?.mdfReqToMDFClms
+										);
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx:111]:
```suggestion
									if (isConfirmed) {
										const newRequestStatus = await patchRequestStatus(
											Status.CANCELED,
											String(row[MDFColumnKey.ID])
										);
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts (hunk 3 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts:19]: Tried to do this, but got this error after change.
image (1)

**Comment by liferay-customer-delivery** [comment]
ci:test:sf

**Comment by abfranca** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/activities-list-panel/index.html [hunk 9]
```diff
@@ -2,7 +2,7 @@
 [#setting date_format="yyyy-MM-dd" /]
 
 [#assign claim_status_style = { "claimed": "ml-3 label label-tonal-info ml-2", "unclaimed": "ml-3 label label-tonal-warning ml-2" } /]
-[#assign request_status_style = { "active": "label label-tonal-info ml-2", "submitted": "label label-tonal-warning ml-2", "approved": "label label-tonal-success ml-2", "expired": "label label-tonal-danger ml-2" } /]
+[#assign request_status_style = { "active": "label label-tonal-info ml-2", "submitted": "label label-tonal-warning ml-2", "approved": "label label-tonal-success ml-2", "expired": "label label-tonal-danger ml-2", "canceled": "label label-tonal-danger ml-2" } /]
```
[the remark this answers is not in the document]
Comment (abfranca) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/activities-list-panel/index.html]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/getDropdownOptions.ts [hunk 10]
```diff
@@ -181,6 +191,18 @@ export default function getDropdownOptions(
 			}
 		}
 
+		if (mdfRequestStatus?.key === Status.APPROVED.key) {
+			if (currentValue === PermissionActionType.CANCEL) {
```
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/getDropdownOptions.ts]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/MDFRequestManagerStatus.tsx [hunk 11]
```diff
@@ -35,7 +35,7 @@ const MDFRequestManagerStatus = () => {
 		MDFRequestDTO
 	>(
 		mdfRequestId &&
-			`/o/${LiferayAPIs.OBJECT}/${ResourceName.MDF_REQUEST_DXP}/${mdfRequestId}?nestedFields=mdfReqToActs`
+			`/o/${LiferayAPIs.OBJECT}/${ResourceName.MDF_REQUEST_DXP}/${mdfRequestId}?nestedFields=mdfReqToActs%2C%20mdfReqToMDFClms`
```
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/MDFRequestManagerStatus.tsx]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts (hunk 4 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts (hunk 5 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts (hunk 6 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts (hunk 7 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx (hunk 8 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx:116]: Couldn't refactor patchRequestStatus function (see print in message above).

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts (hunk 3 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestManagerStatus/util/patchRequestStatus.ts]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx (hunk 8 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx]: Done!

**Review by kylembischof — APPROVED** [review]

**Review by jairmedeiros — APPROVED** [review]

**Comment by jairmedeiros** [comment]
ci:forward:force

**Comment by jairmedeiros** [comment]
ci:forward:force

**Pull request closed without merging**